### PR TITLE
GlobalId

### PR DESCRIPTION
**Pull request by felixKubicek**
@roschaefer review please

### CODE REVIEW

File: UPtables/features/asp/teacher_unavailibility.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  Szenario: Lehrer hat eine Unverfügbarkeit
+  Angenommen Prof. "Scheffer" ist am Montag um 10:00 Uhr unverfügbar
+  Wenn jetzt nach Stundenplänen gesucht wird
+  Dann gibt es 2 optimale Lösungen
+  
+  Szenario: Lehrer ist komplett Unverfügbar
+  Angenommen Prof. "Scheffer" ist am Montag um 10:00 Uhr unverfügbar
+  Und Prof. "Scheffer" ist am Mittwoch um 10:00 Uhr unverfügbar
+  Wenn jetzt nach Stundenplänen gesucht wird
+  Dann gibt es keine Lösung
```
Comment (roschaefer) [inline UPtables/features/asp/teacher_unavailibility.feature]: @felixKubicek perfect! Just to let you know: You can append additional descriptions if you omit the end-of-line in your cucumber regex (/^blabla$/ -> /^blabla/). Sometimes it makes your cucumber scenarios more vivid.

File: UPtables/app/mixins/global_id.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module GlobalId
+
+  def g_id
+    "#{self.class.model_name.param_key}_#{self.id}"
```
Comment (roschaefer) [inline UPtables/app/mixins/global_id.rb]: @felixKubicek refactored this method. Was there a reason why we had ```read_attribute(:id)```?

File: UPtables/lib/asp/constraints/soft/double_lectures.lp [hunk 3]
```diff
@@ -7,8 +7,11 @@
...
-  not assigned(C,R,D,P-1),
-  not assigned(C,R,D,P+1).
+
+
+joined_time_assignment(C,R,WD,TF,TIME) :- assigned(C,R,WD,TF), timeframe(TF,TIME).
+penalty("DoubleLecture", joined_time_assignment(C,R,WD,TF,TIME), penalty_of_double_lectures) :-
+  course_component(C,_,_,_,1), weekday(WD,_),
+  joined_time_assignment(C,R,WD,TF, TIME),
+  not joined_time_assignment(C,R,WD,_, TIME-1),
+  not joined_time_assignment(C,R,WD,_, TIME+1).
```
Comment (roschaefer) [inline UPtables/lib/asp/constraints/soft/double_lectures.lp]: @felixKubicek I decided to include ```GlobalId``` in ```Timeframe``` and ```Weekday``` as well. The price for this is a performance hit for the grounding in this particular constraint. I have to *join* the time value back into another compound term called ```joined_time_assignment```.

What do you think? Is it worth it for the sake of consistency?

File: UPtables/app/mixins/global_id.rb (hunk 2 above)
Comment (felixKubicek) [inline UPtables/app/mixins/global_id.rb]: @roschaefer, agree: self.id is better

File: UPtables/lib/asp/constraints/soft/double_lectures.lp (hunk 3 above)
Comment (felixKubicek) [inline UPtables/lib/asp/constraints/soft/double_lectures.lp]: @roschaefer difficult choice. i would prefer to leave it inconsistent by returning the id for the timeframe

**Pull request merged**